homeassistant/components/nut/config_flow.py

```diff
@@ -54,7 +54,7 @@ def _resource_schema(available_resources, selected_resources):
 
 def _ups_schema(ups_list):
     """UPS selection schema."""
-    ups_map = {ups: ups for ups in ups_list}
+    ups_map = {ups: name for ups, name in ups_list.items()}
```

Comment:
Could `ups_list` be `None` here now?

Reply:
It should never be None or something is broken with the NUT server.

Reply:
Shouldn't be able to get there as we would have raised CannotConnect because `status` would be `None` in `validate_input`
```
    if not status:
        raise CannotConnect
```

Reply:
It looks like get_alias can fail and status not fail at the same time?

Reply:
Will `self._client.list_vars` blow up if passing `None` as alias?

Reply:
It will blow up with
```2020-04-08 22:18:58 DEBUG (SyncWorker_11) [custom_components.nut] Error getting NUT vars for host localhost: ERR UNKNOWN-UPS
```